tests/unit/api/main.py

```diff
@@ -639,7 +639,7 @@ def test_node_set_metadata_diff_nodes(self):
 
     def test_node_set_metadata_non_string(self):
         api.node_set_metadata('free_node_0', 'JSON',
-                              json.dumps({"val1": 1, "val2": 2}))
+                              '{"val1": 1, "val2": 2}')
```

Comment:
You shouldn't  quote this -- the function accepts *parsed* JSON, so it'll be a dictonary by the time it gets this far in a running server.